CenterForOpenScience/ember-osf-web
# [EMB-493][Collections] "Discard Changes" resets subjects on edit page when the project already has subjects.

## Purpose

Currently on edit (not submit) page of Collections, when a project already has subjects, hitting `Discard changes` button on the subject picker clears all subjects, instead of resetting it to the state prior to changes. This PR fixes the problem.

## Summary of Changes

In `subject-picker/component.ts`, the `discard()` method actually restores `initialSubjects` to  `currentSubjects`. The problem is that we always set `initialSubjects` to be an empty array, regardless of whether the project already has subjects associated with it.
Therefore, in the `didInsertElement()` hook, we store `currentSubjects`, which are the pre-existing subjects passed to the component, to `initialSubjects` so that we can restore it later if needed.

## QA Notes

Make sure that on edit page of Collections,  hitting `Discard changes` would not clear all subjects. Also make sure that `Discard changes` on submit page works as it did prior to the change.

## Ticket

https://openscience.atlassian.net/browse/EMB-493

# Reviewer Checklist
